File: LeetCode Solutions/September 2022 LeetCoding Challenge/Average of Levels in Binary Tree.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def averageOfLevels(self, root: Optional[TreeNode]) -> List[float]:
        queue, averageValue = [root], []
        while(len(queue)):
            queueLength, row = len(queue), 0
            for i in range(queueLength):
                current = queue.pop(0)
                row += current.val
                if(current.left):
                    queue.append(current.left)
                if(current.right):
                    queue.append(current.right)
            averageValue.append(row/queueLength)
        return averageValue

File: LeetCode Solutions/LeetCoding Challenge/March 2021 LeetCoding Challenge/Average of Levels in Binary Tree/Average of Levels in Binary Tree.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */
class Solution {
public:
    vector<double> averageOfLevels(TreeNode* root) {
        queue<TreeNode*> queue;
        queue.push(root);
        vector<double> averageValue;
        while (queue.size()){
            double queueLength = queue.size(), row = 0;
            for (int i=0; i<queueLength; i++){
                TreeNode* current = queue.front(); 
                queue.pop();
                row += current->val;
                if (current->left){
                    queue.push(current->left);
                }
                if (current->right){
                    queue.push(current->right);
                }
            }
            averageValue.push_back(row/queueLength);
        }
        return averageValue;
    }
};